src/Agent.Worker/Container/DockerCommandManager.cs

```diff
@@ -90,8 +119,20 @@ public override void Initialize(IHostContext hostContext)
                 }
             }
 
-            string dockerArgs = $"--name {context.Container.ContainerName} --rm -v /var/run/docker.sock:/var/run/docker.sock {dockerMountVolumesArgs} {image} sleep 999d";
-            return (await ExecuteDockerCommandAsync(context, "create", dockerArgs)).FirstOrDefault();
+#if !OS_WINDOWS
+            string dockerArgs = $"--name {container.ContainerDisplayName} --rm --network={_dockerNetwork} -v /var/run/docker.sock:/var/run/docker.sock {container.ContainerCreateOptions} {dockerMountVolumesArgs} {container.ContainerImage} sleep 999d";
```

Comment:
use `node -e "setInterval(function(){}, 24*60*60*1000);"` in case sleep doesn't exist in some linux image.